elastic/apm-server
# sampling: require a default policy

## Motivation/summary

When tail-based sampling is enabled, require a default policy to be defined. A default policy is one with no criteria, only a sampling rate which applies to any traces that does not match any other policy.

Without a default policy, the default behaviour is to drop traces that do not match any policy. By requiring a default we ensure that this behaviour is intentional: either drop (sample_rate=0) or not (sample_rate>0).

## Checklist

~- [ ] Documentation has been updated~ (no docs yet)

## How to test these changes

Run apm-server with `-E apm-server.sampling.tail.enabled`, check that it exits with an error because no default policy is defined.

## Related issues

Closes #4583
